pkg/reconciler/build/build.go

```diff
@@ -235,10 +226,16 @@ func (c *Reconciler) Reconcile(ctx context.Context, key string) error {
 }
 
 func (c *Reconciler) updateStatus(u *v1alpha1.Build) error {
+	buildStatus.Lock()
```

Comment:
I'd like to avoid this global lock across all build status updates. Can we get away with locking around each build that needs to be updated instead? e.g., Use a `map[string]sync.Mutex` where the key is the build key (namespace+name)?